lib/Service/RoomService.php

```diff
@@ -391,4 +404,121 @@ public function verifyPassword(Room $room, string $password): array {
 			'url' => '',
 		];
 	}
+
+	public function setTimeToLive(Room $room, string $userId, int $ttl): void {
+		$event = new ChangeTtlEvent($room, $ttl);
+		$this->dispatcher->dispatch(Room::EVENT_BEFORE_SET_TIME_TO_LIVE, $event);
+
+		$update = $this->db->getQueryBuilder();
+		$update->update('talk_rooms')
+			->set('time_to_live', $update->createNamedParameter($ttl, IQueryBuilder::PARAM_INT))
+			->where($update->expr()->eq('id', $update->createNamedParameter($room->getId(), IQueryBuilder::PARAM_INT)));
+		$update->executeStatement();
+		$jobList = Server::get(IJobList::class);
+		if ($ttl > 0) {
+			$this->ttlSystemMessage($room, $userId, $ttl, 'ttl_enabled');
+			$jobList->add(ApplyTtl::class, ['room_id' => $room->getId()]);
+		} else {
+			$this->ttlSystemMessage($room, $userId, $ttl, 'ttl_disabled');
+			$jobList->remove(ApplyTtl::class, ['room_id' => $room->getId()]);
+		}
+
+		$this->dispatcher->dispatch(Room::EVENT_AFTER_SET_TIME_TO_LIVE, $event);
+	}
+
+	private function ttlSystemMessage(Room $room, string $userId, int $ttl, string $message): void {
+		$this->chatManager->addSystemMessage(
+			$room,
+			$room->getParticipant($userId)->getAttendee()->getActorType(),
+			$room->getParticipant($userId)->getAttendee()->getActorId(),
+			json_encode([
+				'message' => $message,
+				'parameters' => ['ttl' => $ttl]
+			]),
+			$this->timeFactory->getDateTime(),
+			false
+		);
+	}
+
+
+	public function deleteExpiredTtl(int $roomId, int $jobId): array {
+		$room = $this->manager->getRoomById($roomId);
+
+		$max = $this->getMaxDateTtl($room->getTimeToLive());
+		$min = $this->getMinDateTtl($jobId);
+
+		$ids = $this->getMessageIdsByRoomIdInDateInterval($roomId, $min, $max);
+		if (count($ids)) {
+			$this->reportDeletedMessagesIds($room, $ids);
+			$this->deleteMessagesByIds($ids);
+		}
+		return $ids;
+	}
+
+	private function getMaxDateTtl(int $ttl): \DateTime {
+		$max = $this->timeFactory->getDateTime();
+		return $max->sub(new DateInterval('PT' . $ttl . 'S'));
+	}
+
+	private function getMinDateTtl(int $jobId): \DateTime {
+		$query = $this->db->getQueryBuilder();
+		$query->select('last_checked')
+			->from('jobs')
+			->where(
+				$query->expr()->eq('id', $query->createNamedParameter($jobId, IQueryBuilder::PARAM_INT))
+			);
+		$result = $query->executeQuery();
+		$lastCheckedEpoch = $result->fetchOne();
+		$lastChechedDateTime = $this->timeFactory->getDateTime('@' . $lastCheckedEpoch);
+		return $lastChechedDateTime;
+	}
+
+	private function getMessageIdsByRoomIdInDateInterval(int $roomId, \DateTime $min, \DateTime $max): array {
+		$query = $this->db->getQueryBuilder();
+		$query->select('id')
+			->from('comments')
```

Comment:
needs to go into `Chat/CommentsManager.php` and should be upstreamed for the release.